pkg/processor/trigger/trigger.go

```diff
@@ -349,17 +349,19 @@ func (at *AbstractTrigger) SubscribeToControlMessageKind(kind controlcommunicati
 func (at *AbstractTrigger) UnsubscribeFromControlMessageKind(kind controlcommunication.ControlMessageKind,
 	controlMessageChan chan *controlcommunication.ControlMessage) error {
 
-	at.Logger.DebugWith("Unsubscribing channel from control message kind",
-		"kind", kind,
-		"numWorkers", len(at.WorkerAllocator.GetWorkers()))
-
 	for _, workerInstance := range at.WorkerAllocator.GetWorkers() {
+		at.Logger.DebugWith("Unsubscribing channel from control message kind",
+			"kind", kind,
+			"worker_id", workerInstance.GetIndex())
 		if err := workerInstance.Unsubscribe(kind, controlMessageChan); err != nil {
 			return errors.Wrapf(err,
 				"Failed to unsubscribe channel from control message kind %s in worker %d",
 				kind,
 				workerInstance.GetIndex())
 		}
+		at.Logger.DebugWith("Successfully unsubscribed channel from control message kind",
+			"kind", kind,
+			"worker_id", workerInstance.GetIndex())
```

Comment:
Wouldn't this be too spammy if we have many workers?..

Reply:
@TomerShor yes, it probably will. Let me rework just for all workers, so we inform that all workers have unsubscribed.

Reply:
@TomerShor While reviewing this method, I noticed another potential bug. We currently return on error, which means we won't unsubscribe from other workers' brokers. This could also result in multiple panics. Although it's unlikely to encounter this scenario because we can only have an error if the consumer doesn't exist, I'll still fix it as a precaution.